core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -700,4 +702,176 @@ class AdminManager(val config: KafkaConfig,
     val readOnly = !DynamicBrokerConfig.AllDynamicConfigs.contains(name)
     new DescribeConfigsResponse.ConfigEntry(name, valueAsString, source, isSensitive, readOnly, synonyms.asJava)
   }
+
+  private def entityToSanitizedUserClientId(entity: QuotaEntity): (Option[String], Option[String]) = {
+    if (entity.entries.isEmpty)
+      throw new InvalidRequestException("Invalid empty quota entity")
+
+    var user: Option[String] = None
+    var clientId: Option[String] = None
+    entity.entries().asScala.foreach { case (entityType, entityName) =>
+      val sanitizedEntityName = Some(Sanitizer.sanitize(entityName))
+      entityType match {
+        case QuotaEntity.USER => user = sanitizedEntityName
+        case QuotaEntity.CLIENT_ID => clientId = sanitizedEntityName
+        case _ => throw new InvalidRequestException(s"Unhandled quota entity type: ${entityType}")
+      }
+      if (entityName == "")
+        throw new InvalidRequestException(s"Empty ${entityType} not supported")
+    }
+    (user, clientId)
+  }
+
+  private def userClientIdToEntity(user: Option[String], clientId: Option[String]): QuotaEntity = {
+    new QuotaEntity((user.map(u => QuotaEntity.USER -> u) ++ clientId.map(c => QuotaEntity.CLIENT_ID -> c)).toMap.asJava)
+  }
+
+  def describeClientQuotas(filters: Seq[QuotaFilter], includeUnspecifiedTypes: Boolean): Map[QuotaEntity, Map[String, Double]] = {
+    var userFilter: Option[QuotaFilter] = None
+    var clientIdFilter: Option[QuotaFilter] = None
+    var otherFilter: Boolean = false
+    filters.foreach { filter =>
+      filter.entityType() match {
+        case QuotaEntity.USER =>
+          if (userFilter.isDefined)
+            throw new InvalidRequestException(s"Duplicate user filter entity type");
+          userFilter = Some(filter)
+        case QuotaEntity.CLIENT_ID =>
+          if (clientIdFilter.isDefined)
+            throw new InvalidRequestException(s"Duplicate client filter entity type");
+          clientIdFilter = Some(filter)
+        case "" =>
+          throw new InvalidRequestException(s"Unexpected empty filter entity type")
+        case _ =>
+          // Supplying other entity types is not yet supported.
```

Comment:
If it's not supported, it shouldn't silently return incorrect data.  It should throw an `UnsupportedVersionException`

Reply:
Done.